src/services/analytics.js

```diff
@@ -1,5 +1,23 @@
 const getAnalyticsData = (startDate, endDate) => {
-  return Fliplet.Organizations.Analytics.get({ startDate, endDate });
+  const key = 'org-dashboard-date-range';
+
+  startDate = moment(startDate).toISOString();
+  endDate = moment(endDate).toISOString();
+
+  return Fliplet.Storage.get(key).then((value = {}) => {
+    return Fliplet.Cache.get({
+      key: 'org-analytics-cache',
+      expire: 60 * 60 * 24,
+      allowInterfaceCache: true,
+      forceBackgroundUpdate: value.startDate !== startDate || value.endDate !== endDate
+    }, () => {
+      return Fliplet.Organizations.Analytics.get({ startDate, endDate });
+    }).then((result) => {
+      return Fliplet.Storage.set(key, { startDate, endDate }).then(() => {
```

Comment:
How much data are we talking about? Just checking it's not too large to be cached.

Reply:
@tonytlwu thanks, I can see that the payload is relatively small so I'm good with that.

In regards to optimising this in the backend, can you send me the request body and full endpoint (including appId and environment) for that request so that I can look into it?

Reply:
Added to Jira as comment